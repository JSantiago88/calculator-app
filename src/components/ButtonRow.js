import React from 'react';

import { Button } from "./Button";

export const ButtonRow = ({ elements }) => {

    return (
        <div className="button-row">
            {
                elements.map((element) => (
                    <Button
                        key={element.value}
                        values={element}
                    />
                ))
            }
        </div>
    );
};