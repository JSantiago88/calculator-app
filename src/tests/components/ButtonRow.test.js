import React from 'react';
import { shallow } from 'enzyme';

import ButtonRow from '../../components/ButtonRow';

describe('Test in <ButtonRow />', () => {

    const buttons = [];
    const handleSelectButton = () => { };

    const wrapper = shallow(<ButtonRow
        buttons={buttons}
        handleSelectButton={handleSelectButton}
    />);

    test('should ButtonRow the component correctly', () => {

        expect(wrapper).toMatchSnapshot();
    });


    test('should have a css class in the div', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('button-row')).toBe(true);
    });

});
