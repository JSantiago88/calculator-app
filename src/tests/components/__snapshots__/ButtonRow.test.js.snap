// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ButtonRow /> should ButtonRow the component correctly 1`] = `
<div
  className="button-row"
/>
`;
